src/Datagrid/ProxyQuery.php

```diff
@@ -177,6 +177,7 @@ public function getDoctrineQuery(): Query
             }
         }
 
+        /** @var class-string|false $rootEntity */
         $rootEntity = current($queryBuilder->getRootEntities());
```

Comment:
Should we provide a PR to doctrine/orm to update the phpdoc of https://github.com/doctrine/orm/blob/2.12.x/lib/Doctrine/ORM/QueryBuilder.php#L507 then ?

Reply:
Probably yes but it needs some changes related to that method.

Should we go with this solution or with a supress?

Reply:
A suppress would trigger a failure as soon as the pr is merged/released on orm.
So it seems better because with the phpdoc solution we will forget to remove the phpdoc.

Do you agree ?

Reply:
I agree, better to suppress. Can you do it @dmaicher ?

Reply:
You will also need to supress psalm report